iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -585,7 +586,7 @@ private async Task HandleDisconnectAsync()
 
                     await base.OpenAsync(cancellationToken).ConfigureAwait(false);
 
-                    var tasks = new List<Task>(4);
+                    var tasks = new List<Task>(3);
```

Comment:
nit - q: is there a reason why we need the size to be defined in advance?

Reply:
So it is pre-allocated for the size we know it needs to be. It just reduces memory reallocations or overallocations. Do we need it? No. Is it better for memory usage? Yes.